heron/tools/tracker/src/python/query_operators.py

```diff
@@ -460,14 +460,15 @@ async def execute(self, tracker, tmanager: TManagerLocation, start: int, end: in
       for key, metric in metrics2.items():
         # Initialize with first metrics timeline, but second metric's instance
         # because that is multivariate
-        met = Metrics(None, None, metric.instance, start, end, metrics[""].timeline.copy())
-        for timestamp in list(met.timeline.keys()):
-          v = self._f(met.timeline[timestamp], metric.timeline.get(timestamp))
-          if v is None:
-            met.timeline.pop(timestamp, None)
-          else:
-            met.timeline[timestamp] = v
-        all_metrics.append(met)
+        if metrics:
+          met = Metrics(None, None, metric.instance, start, end, metrics[""].timeline.copy())
+          for timestamp in list(met.timeline.keys()):
+            v = self._f(met.timeline[timestamp], metric.timeline[timestamp])
```

Comment:
The method `.get` in `dict` is safer than the brackets because it will return `None` if the key is not found. This is currently raising a `KeyError` in the test suite. Reverting.